SendToRecvFromAsync_Single_Datagram_UDP_IPv6 failed in CI on Windows

http://dotnet-ci.cloudapp.net/job/dotnet_corefx_outerloop_windows_debug/721/console
```
Discovering: System.Net.Sockets.Tests
  Discovered:  System.Net.Sockets.Tests
  Starting:    System.Net.Sockets.Tests
     System.Net.Sockets.Tests.SendReceive.SendToRecvFromAsync_Single_Datagram_UDP_IPv6 [FAIL]
        Assert.Equal() Failure
        Expected: 0
        Actual:   2037238909
        Stack Trace:
           d:\j\workspace\dotnet_corefx_outerloop_windows_debug\src\System.Net.Sockets\tests\FunctionalTests\SendReceive.cs(124,0): at System.Net.Sockets.Tests.SendReceive.SendToRecvFromAsync_Datagram_UDP(IPAddress leftAddress, IPAddress rightAddress)
           d:\j\workspace\dotnet_corefx_outerloop_windows_debug\src\System.Net.Sockets\tests\FunctionalTests\SendReceive.cs(315,0): at System.Net.Sockets.Tests.SendReceive.SendToRecvFromAsync_Single_Datagram_UDP_IPv6()
  Finished:    System.Net.Sockets.Tests
  
  === TEST EXECUTION SUMMARY ===
     System.Net.Sockets.Tests  Total: 6, Errors: 0, Failed: 1, Skipped: 0, Time: 2.399s
```
cc: @davidsh, @cipop, @pgavlin

Another failure in outerloop on Windows that looks by name to be very similar:
http://dotnet-ci.cloudapp.net/job/dotnet_corefx_outerloop_windows_release/719/consoleFull
```
 Unhandled Exception: Xunit.Sdk.NullException: Assert.Null() Failure
  Expected: (null)
  Actual:   0
     at System.Net.Sockets.Tests.SendReceive.<>c__DisplayClass5.<SendToRecvFromAPM_Datagram_UDP>b__3(Int32 received, EndPoint remote)
     at System.Net.LazyAsyncResult.Complete(IntPtr userToken)
     at System.Net.ContextAwareResult.Complete(IntPtr userToken)
     at System.Net.LazyAsyncResult.ProtectedInvokeCallback(Object result, IntPtr userToken)
     at System.Net.Sockets.BaseOverlappedAsyncResult.CompletionPortCallback(UInt32 errorCode, UInt32 numBytes, NativeOverlapped* nativeOverlapped)
     at System.Threading._IOCompletionCallback.PerformIOCompletionCallback(UInt32 errorCode, UInt32 numBytes, NativeOverlapped* pOVERLAP)
d:\j\workspace\dotnet_corefx_outerloop_windows_release\packages\Microsoft.DotNet.BuildTools.1.0.25-prerelease-00102\lib\tests.targets(129,5): warning MSB3073: The command "CoreRun.exe xunit.console.netcore.exe System.Net.Sockets.APMServer.Tests.dll  -xml testResults.xml -notrait category=nonwindowstests  -trait category=InnerLoop -trait category=OuterLoop  -notrait category=failing " exited with code -532462766. [d:\j\workspace\dotnet_corefx_outerloop_windows_release\src\System.Net.Sockets.Legacy\tests\FunctionalTests\System.Net.Sockets.APMServer.Tests.csproj]
d:\j\workspace\dotnet_corefx_outerloop_windows_release\packages\Microsoft.DotNet.BuildTools.1.0.25-prerelease-00102\lib\tests.targets(136,5): error : One or more tests failed while running tests from 'System.Net.Sockets.APMServer.Tests' please check log for details! [d:\j\workspace\dotnet_corefx_outerloop_windows_release\src\System.Net.Sockets.Legacy\tests\FunctionalTests\System.Net.Sockets.APMServer.Tests.csproj]
```

--- a/src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.Windows.cs
+++ b/src/System.Net.Sockets/src/System/Net/Sockets/SocketAsyncEventArgs.Windows.cs
@@ -837,21 +837,25 @@ namespace System.Net.Sockets
                 if (_socketAddressGCHandle.IsAllocated)
                 {
                     _socketAddressGCHandle.Free();
+                    _pinnedSocketAddress = null;
                 }
 
                 if (_wsaMessageBufferGCHandle.IsAllocated)
                 {
                     _wsaMessageBufferGCHandle.Free();
+                    _ptrWSAMessageBuffer = IntPtr.Zero;
                 }
 
                 if (_wsaRecvMsgWSABufferArrayGCHandle.IsAllocated)
                 {
                     _wsaRecvMsgWSABufferArrayGCHandle.Free();
+                    _ptrWSARecvMsgWSABufferArray = IntPtr.Zero;
                 }
 
                 if (_controlBufferGCHandle.IsAllocated)
                 {
                     _controlBufferGCHandle.Free();
+                    _ptrControlBuffer = IntPtr.Zero;
                 }
             }
         }

--- a/src/System.Net.Sockets/tests/FunctionalTests/SendReceive.cs
+++ b/src/System.Net.Sockets/tests/FunctionalTests/SendReceive.cs
@@ -449,28 +449,24 @@ namespace System.Net.Sockets.Tests
             Assert.Equal(sentChecksum.Sum, receivedChecksum.Sum);
         }
 
-        [ActiveIssue(3610)]
         [Fact]
         public void SendToRecvFromAsync_Single_Datagram_UDP_IPv6()
         {
             SendToRecvFromAsync_Datagram_UDP(IPAddress.IPv6Loopback, IPAddress.IPv6Loopback);
         }
 
-        [ActiveIssue(3610)]
         [Fact]
         public void SendToRecvFromAsync_Single_Datagram_UDP_IPv4()
         {
             SendToRecvFromAsync_Datagram_UDP(IPAddress.Loopback, IPAddress.Loopback);
         }
 
-        [ActiveIssue(3610)]
         [Fact]
         public void SendToRecvFromAsync_UdpClient_Single_Datagram_UDP_IPv6()
         {
             SendToRecvFromAsync_UdpClient_Datagram_UDP(IPAddress.IPv6Loopback, IPAddress.IPv6Loopback);
         }
 
-        [ActiveIssue(3610)]
         [Fact]
         public void SendToRecvFromAsync_UdpClient_Single_Datagram_UDP_IPv4()
         {

